File: androlua/src/main/assets/lua_requires/executor.lua
--
-- @author Vove
-- 2018/8/15 21:45
-- executor.lua
--

-- require 'import'
--
runtime = executor

actionCount = executor.actionCount
currentActionIndex = executor.currentActionIndex

function accessibility()
    return executor.checkAccessibilityService(true)
end

function requireAccessibility( ... )
    local ps = { ... }
    if #ps == 0 then
        executor.requireAccessibility()
    else
        executor.requireAccessibility(ps[1])
    end
end

function waitAccessibility( ... )
    local ps = { ... }
    if #ps == 0 then
        return executor.waitAccessibility()
    else
        return executor.waitAccessibility(ps[1])
    end
end

function checkService()
    return executor.checkAccessibilityService(false)
end

--  语音合成同步 ，说完再向下执行
-- @param text
-- @return booleam 是否成功
--
function speakSync(text)
    return executor.speakSync(text)
end

--
-- 语音合成
--
function speak(text)
    executor.speak(text)
end

--
--
-- return boolean
function alert(t, m)
    return executor.alert(t, m)
end

--
--
-- return String?
function singleChoiceDialog(title, choices)
    return executor.singleChoiceDialog(title, choices)
end

--
--
-- return string
function waitForVoiceParam()
    return executor.waitForVoiceParam(nil)
end

--
-- s:询问时显示文字
-- return string
function waitForVoiceParam(s)
    return executor.waitForVoiceParam(s)
end

--
-- 等待界面 pkg , activity  millis
-- return boolean
function waitForApp(pkg, ...)
    local ps = { ... }
    local activity
    local millis = 5000
    if (#ps == 1) then
        if (type(ps[1]) == 'string') then
            activity = ps[1]
        elseif (type(ps[1]) == 'number') then
            millis = ps[1]
        end
    elseif (#ps == 2) then
        activity = ps[1]
        millis = ps[2]
    end
    log(activity)
    log(millis)
    return executor.waitForApp(pkg, activity, millis)
end


--[[
-- return ViewNode? , null if AccessibilityService is not running
 ]]
function waitForId(id, ...)
    local ps = { ... }
    local millis = 3000
    if (#ps == 1) then
        millis = ps[1]
    end
    return executor.waitForViewId(id, millis)
end


--[[
-- return ViewNode? , null if AccessibilityService is not running
 ]]
function waitForDesc(desc, ...)
    local ps = { ... }
    local millis = 3000
    if (#ps == 1) then
        millis = ps[1]
    end
    return executor.waitForDesc(desc, millis)
end

--[[
-- return ViewNode? , null if AccessibilityService is not running
 ]]
function waitForText(text, ...)
    local ps = { ... }
    local millis = 3000
    if (#ps == 1) then
        millis = ps[1]
    end
    return executor.waitForText(text, millis)
end


--[[
-- smart smartOpen
 ]]
function smartOpen(s)
    return executor.smartOpen(s)
end
function smartClose(s)
    return executor.smartClose(s)
end

function sleep(m)
    executor.sleep(m)
end

function removeFloat()
    executor.removeFloat()
end

File: rhino-android/src/main/assets/rhino_require/executor.js

actionCount = executor.actionCount
currentActionIndex = executor.currentActionIndex

checkService = function(){
    return executor.checkAccessibilityService(false)
}

notifyFailed = function(s){
    executor.executeFailed(s)
}

accessibility = function(){
    return executor.checkAccessibilityService(true)
}
requireAccessibility = function(){
    executor.requireAccessibility()
}
waitAccessibility = function(m) {
    if(m)
        return executor.waitAccessibility(m)
    else
        return executor.waitAccessibility()
}

//  语音合成同步 ，说完再向下执行
// @param text
// @return booleam 是否成功
//
speakSync = function(text){
    return executor.speakSync(text)
}


//
// 语音合成
//
speak = function(text){
    executor.speak(text)
}


//
//
// return boolean
alert = function(t, m){
    return executor.alert(t, m)
}


//
//
// return String?
singleChoiceDialog = function(title, choices){
    return executor.singleChoiceDialog(title, choices)
}


//
//
// return string
waitForVoiceParam = function(){
    return executor.waitForVoiceParam(nil)
}

//
// s:询问时显示文字
// return string
waitForVoiceParam = function(s){
    return executor.waitForVoiceParam(s)
}

//
// 等待界面 pkg , activity  millis
// return boolean
waitForApp = function(pkg, activity, millis){
    if(millis == undefined) millis = -1
    log(activity)
    log(millis)
    return executor.waitForApp(pkg, activity, millis)
}


//
// return ViewNode? , null if AccessibilityService is not running
 
waitForId = function(id, millis){
    if(millis == undefined) millis = -1
    return executor.waitForViewId(id, millis)
}


//[[
// return ViewNode? , null if AccessibilityService is not running
 
waitForDesc=function(desc,millis){
    if(millis == undefined) millis = -1
    return executor.waitForDesc(desc, millis)
}

//[[
// return ViewNode? , null if AccessibilityService is not running
 
waitForText = function(text, millis){
    if(millis == undefined) millis = -1
    return executor.waitForText(text, millis)
}


//[[
// smart smartOpen
 
smartOpen = function(s){
    return executor.smartOpen(s)
}
smartClose = function(s){
    return executor.smartClose(s)
}
removeFloat = function() {
    executor.removeFloat()
}
sleep = function(m){
    executor.sleep(m)
}


